control-plane/subcommand/gateway-resources/command.go

```diff
@@ -399,6 +494,44 @@ func forceClass(ctx context.Context, k8sClient client.Client, o *gwv1beta1.Gatew
 	}, exponentialBackoffWithMaxIntervalAndTime())
 }
 
+func forceV2ClassConfig(ctx context.Context, k8sClient client.Client, o *v2beta1.GatewayClassConfig) error {
+	return backoff.Retry(func() error {
+		var existing v2beta1.GatewayClassConfig
+		err := k8sClient.Get(ctx, client.ObjectKeyFromObject(o), &existing)
+		if err != nil && !k8serrors.IsNotFound(err) {
+			return err
+		}
+
+		if k8serrors.IsNotFound(err) {
+			return k8sClient.Create(ctx, o)
+		}
+
+		existing.Spec = *o.Spec.DeepCopy()
+		existing.Labels = o.Labels
+
+		return k8sClient.Update(ctx, &existing)
+	}, exponentialBackoffWithMaxIntervalAndTime())
+}
+
+func forceV2Class(ctx context.Context, k8sClient client.Client, o *v2beta1.GatewayClass) error {
+	return backoff.Retry(func() error {
+		var existing v2beta1.GatewayClass
+		err := k8sClient.Get(ctx, client.ObjectKeyFromObject(o), &existing)
+		if err != nil && !k8serrors.IsNotFound(err) {
+			return err
+		}
+
+		if k8serrors.IsNotFound(err) {
+			return k8sClient.Create(ctx, o)
+		}
+
+		existing.Spec = *o.Spec.DeepCopy()
```

Comment:
we need to DeepCopy here to avoid copying the mutex that proto's have on them

Reply:
Is this still required?